bin/setup

```diff
@@ -1,29 +1,65 @@
 #!/usr/bin/env ruby
-require 'pathname'
+require "pathname"
+require "rake"
+
+include FileUtils
+
+ENV["RAILS_ENV"]               ||= ENV["PASSENGER_APP_ENV"] || "development"
+ENV["RAILS_RELATIVE_URL_ROOT"] ||= ENV["PASSENGER_BASE_URI"]
+
+# if relative url root not set, use default for this app
+ENV['RAILS_RELATIVE_URL_ROOT'] ||= "pun/usr/wiag/vftwebapp" if ENV['RAILS_ENV'] == 'production'
+
+# define application object
+class App
+  attr_accessor :env, :url
+
+  def initialize(env:, url:)
+    @env    = env
+    @url    = url
+  end
+
+  def production?
+    env == "production"
+  end
+end
+
+# set application settings
+APP = App.new(
+  env:    ENV["RAILS_ENV"],
+  url:    ENV["RAILS_RELATIVE_URL_ROOT"],
+)
 
 # path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
+APP_ROOT = Pathname.new File.expand_path("../../",  __FILE__)
 
-Dir.chdir APP_ROOT do
+chdir APP_ROOT do
   # This script is a starting point to setup your application.
   # Add necessary setup steps to this file:
 
-  puts "== Installing dependencies =="
-  system "gem install bundler --conservative"
-  system "bundle check || bundle install"
+  puts "\n== Building App =="
+  puts "RAILS_ENV               = #{APP.env}"
+  puts "RAILS_RELATIVE_URL_ROOT = #{APP.url    || "not set"}"
 
-  # puts "\n== Copying sample files =="
-  # unless File.exist?("config/database.yml")
-  #   system "cp config/database.yml.sample config/database.yml"
-  # end
+  unless system("bin/bundle check &> /dev/null")
+    puts "\n== Installing dependencies =="
+    sh "bin/bundle install --path=vendor/bundle"
+    sh "bin/bundle clean"
+  end
 
-  puts "\n== Preparing database =="
-  system "bin/rake db:setup"
+  if APP.production?
+    puts "\n== Compiling assets =="
+    sh "bin/rake assets:clobber"
+    sh "bin/rake assets:precompile"
+  end
 
+  puts "\n== Preparing database =="
+  sh "bin/rake db:setup"
```

Comment:
Remove this section that creates the database. For developers that should be done manually. For production, per user databases are created with the bin/setup-production script.